orders-accounting/src/price_calculation.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common::{
+    chain::{output_value::OutputValue, OrderId},
+    primitives::Amount,
+};
+use utils::ensure;
+
+use crate::{error::Result, Error, OrdersAccountingView};
+
+pub fn calculate_fill_order(
```

Comment:
Please make sure you have tests that incrementally fill an order until it's empty. I still haven't checked the tests, so this may or may not be there.

Reply:
here is the test:
https://github.com/mintlayer/mintlayer-core/blob/1feed1126346c9c4aa14a593da070adf4548a0bb/orders-accounting/src/tests/operations.rs#L523